docs/java-rest/high-level/apis/search.asciidoc

```diff
@@ -1,4 +1,119 @@
 [[java-rest-high-search]]
 === Search API
 
-To be documented.
+[[java-rest-high-document-search-request]]
+==== Search Request
+
+The `SearchRequest` is used for any operation that has to do with searching
+documents, aggregations, suggestions and also offers ways of requesting
+highlighting on the resulting documents.
+
+In its most basic form, a query can be added to the request like this:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-basic]
+--------------------------------------------------
+
+<1> Creates the `SeachRequest`. Without arguments this runs against all indices.
+<2> Most parameters of the search can be added to the `SearchSourceBuilder` which contains everything that
+in the Rest API would be placed in the search request body.
+<3> Add a `match_all` query to the `SearchSourceBuilder`.
+
+==== Optional request arguments
+
+Lets first look at some of the optional argument of a `SearchRequest`.
+First of all, the request can be restricted to one or more indices or types in the following way:
+
+["source","java",subs="attributes,callouts,macros"]
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-request-indices-types]
+--------------------------------------------------
+
+Also, if necessary, a routing parameter can be set:
```

Comment:
do it here like we did it in other pages with the reference in the snippet, that gets visualized after it, rather than the text before it?